tests/pytests/unit/grains/test_core.py

```diff
@@ -3623,3 +3871,1154 @@ def _mock_is_file(filename):
+            assert ret == {"mem_total": 1023, "swap_total": 0}
+
+    osdata = {"kernel": "NetBSD"}
+
+    with patch("salt.utils.path.which", side_effect=["/sbin/sysctl", "/sbin/swapctl"]):
+
+        mock_cmd_run = [
+            "1073278976",
+            "total: 1048559 KBytes allocated, 0 KBytes used, 1048559 KBytes available",
+        ]
+
+        with patch.dict(
+            core.__salt__,
+            {"cmd.run": MagicMock(side_effect=mock_cmd_run)},
+        ):
+            ret = core._bsd_memdata(osdata)
+            assert ret == {"mem_total": 1023, "swap_total": 0}
+
+    with patch("salt.utils.path.which", side_effect=["/sbin/sysctl", "/sbin/swapctl"]):
+
+        mock_cmd_run = [
+            "-",
+            "1073278976",
+            "total: 1048559 KBytes allocated, 0 KBytes used, 1048559 KBytes available",
+        ]
+
+        with patch.dict(
+            core.__salt__,
+            {"cmd.run": MagicMock(side_effect=mock_cmd_run)},
+        ):
+            ret = core._bsd_memdata(osdata)
+            assert ret == {"mem_total": 1023, "swap_total": 0}
+
+    with patch("salt.utils.path.which", side_effect=["/sbin/sysctl", "/sbin/swapctl"]):
+
+        mock_cmd_run = ["-", "1073278976", "no swap devices configured"]
+
+        with patch.dict(
+            core.__salt__,
+            {"cmd.run": MagicMock(side_effect=mock_cmd_run)},
+        ):
+            ret = core._bsd_memdata(osdata)
+            assert ret == {"mem_total": 1023, "swap_total": 0}
+
+    with patch("salt.utils.path.which", side_effect=["/sbin/sysctl", "/sbin/swapctl"]):
+
+        mock_cmd_run = ["-", "1073278976", "no swap devices configured"]
+
+        with patch.dict(
+            core.__salt__,
+            {"cmd.run": MagicMock(side_effect=mock_cmd_run)},
+        ):
+            ret = core._memdata(osdata)
+            assert ret == {"mem_total": 1023, "swap_total": 0}
+
+
+def test__ps():
+    """
+    test _ps
+    """
+    osdata = {"os_family": ""}
+
+    for bsd in ["FreeBSD", "NetBSD", "OpenBSD", "MacOS"]:
+        osdata = {"os": bsd}
+        ret = core._ps(osdata)
+        assert ret == {"ps": "ps auxwww"}
+
+    osdata = {"os_family": "Solaris", "os": ""}
+    ret = core._ps(osdata)
+    assert ret == {"ps": "/usr/ucb/ps auxwww"}
+
+    osdata = {"os": "Windows", "os_family": ""}
+    ret = core._ps(osdata)
+    assert ret == {"ps": "tasklist.exe"}
+
+    osdata = {"os": "", "os_family": "AIX"}
+    ret = core._ps(osdata)
+    assert ret == {"ps": "/usr/bin/ps auxww"}
+
+    osdata = {"os": "", "os_family": "NILinuxRT"}
+    ret = core._ps(osdata)
+    assert ret == {"ps": "ps -o user,pid,ppid,tty,time,comm"}
+
+    osdata = {"os": "", "os_family": "", "virtual": "openvzhn"}
+    ret = core._ps(osdata)
+    assert ret == {
+        "ps": (
+            'ps -fH -p $(grep -l "^envID:[[:space:]]*0\\$" '
+            '/proc/[0-9]*/status | sed -e "s=/proc/\\([0-9]*\\)/.*=\\1=")  '
+            "| awk '{ $7=\"\"; print }'"
+        )
+    }
```

Comment:
Ideally, I'd also love to see these 5K lines of test code to be split into several modules if we can somehow find some king of grouping to them... Not a hard change request from the either, but a nice to have.